src/praisonai-agents/test_async_sequential.py

```
@@ -0,0 +1,222 @@
+        print(f"\nTotal execution time: {end_time - start_time:.2f}s")
+        
+        # Analyze results
+        print("\n=== Execution Analysis ===")
+        
+        # Check if first batch of async tasks ran in parallel
+        async_batch1 = [log for log in execution_log if log["task"] in ["async_task_1", "async_task_2", "async_task_3"]]
+        if async_batch1:
+            earliest_start = min(log["start"] for log in async_batch1)
+            latest_end = max(log["end"] for log in async_batch1)
+            batch1_duration = latest_end - earliest_start
+            print(f"\nFirst async batch (3 tasks, 2s each):")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)